src/Namotion.Reflection/XmlDocsFormatting.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Xml.Linq;
+
+namespace Namotion.Reflection
+{
+    /// <summary>
+    /// Contains the logic to maintain formatting of XML-doc elements.
+    /// </summary>
+    internal static class XmlDocsFormatting
+    {
+        private static readonly IDictionary<XmlDocsFormattingMode, Func<StringBuilder, XElement, StringBuilder>> formattingFunctions =
+            new Dictionary<XmlDocsFormattingMode, Func<StringBuilder, XElement, StringBuilder>>()
+            {
+                { XmlDocsFormattingMode.Unformatted, AppendUnformattedElement },
+                { XmlDocsFormattingMode.Html, AppendHtmlFormattedElement },
+                { XmlDocsFormattingMode.Markdown, AppendMarkdownFormattedElement }
+            };
+
+        /// <summary>
+        /// Appends the value of <paramref name="element"/> to the <paramref name="stringBuilder"/> respecting
+        /// <paramref name="formattingMode"/> to generate additional formatting information.
+        /// </summary>
+        /// <param name="stringBuilder">The current <see cref="StringBuilder"/>.</param>
+        /// <param name="element">The <see cref="XElement"/> to append.</param>
+        /// <param name="formattingMode">The <see cref="XmlDocsFormattingMode"/> for generating additional formatting tags.</param>
+        /// <returns>The passed in <paramref name="stringBuilder"/>.</returns>
+        public static StringBuilder AppendFormattedElement(this StringBuilder stringBuilder, XElement element, XmlDocsFormattingMode formattingMode)
+        {
+            // call apropriate formatting function
+            return formattingFunctions[formattingMode](stringBuilder, element);
+        }
+
+        #region No formatting
```

Comment:
nit: I personally am not a fan of regions

Reply:
Removed `#region`